File: clojure/word-count/word_count.clj
(ns word-count
  (:require [clojure.string :as str]))

(defn- normalize-text [text]
  (str/lower-case (str/replace text #"[^\w]" " ")))

(defn word-count [text]
  (frequencies (str/split (normalize-text text) #"\s+")))

File: javascript/word-count/word-count.js
var Words = function() {};

Words.prototype.count = function(text) {
  var tokens = text.trim().replace(/\s+/, ' ').split(' ');
  return tokens.reduce(function(counts, word) {
    if (isNaN(counts[word])) {
      counts[word] = 0;
    }
    counts[word] += 1;
    return counts;
  }, {});
};

module.exports = Words;
